src/Microsoft.AspNetCore.Server.Kestrel.Https/HttpsConnectionAdapter.cs

```diff
@@ -37,6 +44,11 @@ public HttpsConnectionAdapter(HttpsConnectionAdapterOptions options, ILoggerFact
                 throw new ArgumentException(HttpsStrings.ServiceCertificateRequired, nameof(options));
             }
 
+            // capture the certificate now so it can be switched after validation
+            _serverCertificate = options.ServerCertificate;
+
+            ValidateEku(_serverCertificate);
```

Comment:
They're functionally equivalent, so I don't care either way...but normally we like our "Options" typos to be POCOs, so I opted to put the validation here.